README.md

```diff
@@ -4,6 +4,37 @@
 
 This gem is an implementation of the [language server protocol specification](https://microsoft.github.io/language-server-protocol/) for Ruby, used to improve editor features.
 
+# Overview
+
+The intention of Ruby LSP is to provide a fast, accurate, and robust coding environment for Ruby developers, especially for large codebases.
```

Comment:
Because we don't perform typechecking, our features are actually never 100% accurate, but rather are a best effort. For example, when we implement go to definition, we'll match methods based on name and arity, which is guaranteed to include incorrect results.

```suggestion
The intention of Ruby LSP is to provide a fast, robust and feature rich coding environment for Ruby developers. It does not perform typechecking, so its features are implemented in a best effort basis, attempting to be as accurate as possible.
```